webapp/src/components/CreateEstatePage/EditEstate/EditEstateForm/EditEstateForm.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { Button, Form, Input } from 'semantic-ui-react'
+import { preventDefault } from 'lib/utils'
+import { t } from 'modules/translation/utils'
+import { estateType } from 'components/types'
+
+import './EditEstateForm.css'
+
+export default class EditEstateForm extends React.PureComponent {
+  static propTypes = {
+    value: estateType.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    onCancel: PropTypes.func.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+    const { value: estate } = this.props
+    this.state = {
+      initialEstate: estate
+    }
+  }
+
+  handleNameChange = event => {
+    const { value: estate, onChange } = this.props
+    onChange({
+      ...estate,
+      data: {
+        ...estate.data,
+        name: event.target.value
+      }
+    })
+  }
+
+  handleDescriptionChange = event => {
+    const { value: estate, onChange } = this.props
+    onChange({
+      ...estate,
+      data: {
+        ...estate.data,
+        description: event.target.value
+      }
+    })
+  }
+
+  hasChanged() {
+    const { data } = this.props.value
+    const { name, description } = this.state.initialEstate
+
+    return name !== data.name || description !== data.description
+  }
+
+  render() {
+    const { onCancel, onSubmit, value: estate } = this.props
+    const { name, description } = estate.data
+
+    return (
+      <Form className="EditEstateForm" onSubmit={preventDefault(onSubmit)}>
+        <Form.Field>
+          <label>{t('estate_edit.name')}</label>
+          <Input
+            type="text"
+            value={name}
+            onChange={this.handleNameChange}
+            error={name.length > 50}
```

Comment:
Now that this two hardcoded values (`50` and `140`) are repeated here and on `EditParcelForm` can we abstract them into a method/constant?

Maybe `lib/utils` or `lib/dataUtils` and `isValidName`+`isValidDescription`

Reply:
Good idea! done 72f4a70